packages/api/utils/requestValidation.ts

```diff
@@ -29,3 +32,91 @@ export function shapeValidationForPostRequest<T>(
+ * - 1 application admin role for each application (without a portfolio admin role above)
+ * - 1 environment admin role for each environment (without a portfolio or application admin role above)
+ *
+ * @param applicationStep - the incoming applicationStep request body
+ * @returns - an object summarizing the admin roles found or missing
+ */
+export function findAdministrators(applicationStep: ApplicationStep): AdministratorsFound {
+  const { operators, applications } = applicationStep;
+  const hasPortfolioAdminRole = adminRoleFound(operators);
+
+  // find apps and envs without admin roles
+  const applicationsWithNoAdmin: number[] = [];
+  const environmentsWithNoAdmin: EnvironmentWithNoAdminProps[] = [];
+  applications.forEach((app, appIndex) => {
+    // apps without admin roles
+    !adminRoleFound(app.operators) && applicationsWithNoAdmin.push(appIndex);
+
+    // env without admin roles
+    app.environments.forEach((env, envIndex) => {
+      !adminRoleFound(env.operators) && environmentsWithNoAdmin.push({ appIndex, envIndex });
+    });
+  });
+
+  // if no missing admin roles in app or env, all app or env admins are considered present
+  const hasAdminForEachApplication = applicationsWithNoAdmin.length === 0;
+  const hasAdminForEachEnvironment = environmentsWithNoAdmin.length === 0;
+
+  let acceptableAdministratorRoles = hasPortfolioAdminRole || hasAdminForEachApplication || hasAdminForEachEnvironment;
+
+  if (!acceptableAdministratorRoles) {
+    let appsWithMissingAppAndEnvAdminRole = 0;
+    applicationsWithNoAdmin.forEach((appNoAdmin) => {
+      const missingEnvAdmins = environmentsWithNoAdmin.filter((obj) => obj.appIndex === appNoAdmin);
+      // if no missing env admin from apps with missing app admin, all env admins considered present
+      missingEnvAdmins.length !== 0 && appsWithMissingAppAndEnvAdminRole++;
+    });
+
+    acceptableAdministratorRoles = appsWithMissingAppAndEnvAdminRole === 0;
+  }
+
+  return {
+    hasPortfolioAdminRole,
+    hasAdminForEachApplication,
+    hasAdminForEachEnvironment,
+    applicationsWithNoAdmin,
+    environmentsWithNoAdmin,
+    acceptableAdministratorRoles,
+  };
+}
+
+/**
+ * Checks if at least one operator admin is present in a group of operators
+ * according to the business rules.
+ *
+ * Only 'portfolio_administrator' operators are found at the top level, and
+ * 'administrator' operators are found at the app and env levels.
+ *
+ * @param operators - operators at one of the three application step levels
+ * @returns - true if one operator admin found, and false otherwise
+ */
+export function adminRoleFound(operators: Operator[]): boolean {
+  return operators.some(isAdministrator);
+}
+
+// TODO: move into Operators.ts and update with changes from Operators.ts
+export function isAdministrator(operator: Operator): boolean {
+  return operator.access === AccessLevel.PORTFOLIO_ADMINISTRATOR || operator.access === AccessLevel.ADMINISTRATOR;
+}
```

Comment:
Can we just do this now?

Reply:
Agree.  #387 recently merged, so go ahead an do this now.

Reply:
Thanks for getting that merged @jasonburkert-ccpo. My intent was to do that 
once those changes were merged with the expectation that the merge would 
happen soon and I could do it on this branch.